src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -234,19 +281,20 @@ public override bool TryGetPropertyType(string name, out Type type)
         /// Returns the instance that holds all the changes (and original values) being tracked by this Delta.
         /// </summary>
         [SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate", Justification = "Not appropriate to be a property")]
-        public TStructuralType GetInstance()
+        internal override object GetInstance()
```

Comment:
>internal override object [](start = 8, length = 24)

why change the return type and mark it from public to internal?

Reply:
This change doesn't make sense to me.

---
In reply to: [190749152](https://github.com/OData/webapi/pull/1445#discussion_r190749152) [](ancestors = 190749152)

Reply:
+1

Reply:
It was changed to return object type due to we need to return Delta<> with variable type for nested resources.
Now the relevant code in TryGetPropertyValue has been further updated to using reflection to retrieve the instance value --> Restored to the original API.
Thanks.

---
In reply to: [190749966](https://github.com/OData/WebApi/pull/1445#discussion_r190749966) [](ancestors = 190749966,190749152)